group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -1561,6 +1604,156 @@ public void testConsumerGroupOffsetFetchWithStaleMemberEpoch() {
             () -> context.fetchAllOffsets("group", "member", 10, Long.MAX_VALUE));
     }
 
+    static private void testOffsetDeleteWith(
+        OffsetMetadataManagerTestContext context,
+        String groupId,
+        String topic,
+        int partition,
+        Errors error
+    ) {
+        final OffsetDeleteRequestData.OffsetDeleteRequestTopicCollection requestTopicCollection =
+            new OffsetDeleteRequestData.OffsetDeleteRequestTopicCollection();
+        requestTopicCollection.add(
+            new OffsetDeleteRequestData.OffsetDeleteRequestTopic()
+                .setName(topic)
+                .setPartitions(Collections.singletonList(
+                    new OffsetDeleteRequestData.OffsetDeleteRequestPartition().setPartitionIndex(partition)
+                ))
+        );
+
+        final OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection expectedResponsePartitionCollection =
+            new OffsetDeleteResponseData.OffsetDeleteResponsePartitionCollection();
+        expectedResponsePartitionCollection.add(
+            new OffsetDeleteResponseData.OffsetDeleteResponsePartition()
+                .setPartitionIndex(partition)
+                .setErrorCode(error.code())
+        );
+        final OffsetDeleteResponseData.OffsetDeleteResponseTopicCollection expectedResponseTopicCollection =
+            new OffsetDeleteResponseData.OffsetDeleteResponseTopicCollection();
+        expectedResponseTopicCollection.add(
+            new OffsetDeleteResponseData.OffsetDeleteResponseTopic()
+                .setName(topic)
+                .setPartitions(expectedResponsePartitionCollection)
+        );
+
+        final List<Record> expectedRecords = error == Errors.NONE &&
+            context.offsetMetadataManager.offset(groupId, topic, partition) != null ?
+            Collections.singletonList(
+                RecordHelpers.newOffsetCommitTombstoneRecord(groupId, topic, partition)
+            ) :
+            Collections.emptyList();
```

Comment:
This block is really hard to parse. 

```
final List<Record> expectedRecords = 
     error == Errors.NONE && context.offsetMetadataManager.offset(groupId, topic, partition) != null ?
            Collections.singletonList(RecordHelpers.newOffsetCommitTombstoneRecord(groupId, topic, partition)) :
            Collections.emptyList();
```

Would it be better like this? Otherwise, I would use a regular if statement.